src/modules/ethereum/transactions/interpret.ts

```diff
@@ -8,7 +8,8 @@ import { getGas } from './getGas';
 
 export function interpretTransaction(
   address: string,
-  transaction: IncomingTransactionWithChainId
+  transaction: IncomingTransactionWithChainId,
+  origin?: string
```

Comment:
Maybe let's make it required but nullable? So that it is explicitly omitted
Also maybe worth changing the signature to named params